### DISCUSSION TITLE
rdkit.js feature request: Add attribution information (from atom or bond index) for the items highlighted in the svg

### DISCUSSION THREAD

**Discussion by ShangChien** [rdkit/rdkit#5177] [Development]
**Add attribution information (from atom or bond index) for the elements highlighted in the svg, this will be helpful and convenient for further coding about web interaction**

original discuss: https://github.com/MichelML/rdkit-js/issues/71#issue-1193965423

current highlight element in svg (generated by rdkit.js), which without info of index
![2022-04-06 113237](https://user-images.githubusercontent.com/48350636/162342275-8f75060d-f240-461d-a304-6706137b5a76.png)
current solid line element in svg (generated by rdkit.js), which with info of index in class
![2022-04-06 113306](https://user-images.githubusercontent.com/48350636/162342782-4857523f-f96d-4b5d-977e-c8f7b01b3e9b.png)

**desired features:**
When rdkit returns the svg, for each highlighted item, providing the index (bond or atom) as the solid line item.

I don't know if it is possible to modify the current wasm to achieve my idea.
Thanks for all developers.

**Comment by greglandrum** [comment]
@ShangChien : though I can reproduce the problem with missing class tags for atom highlights from Python with v2021.09.x of the RDKit, I *do* get highlights when using v2022.03.1.
Which version of the rdkit.js are you using?

**Reply by ShangChien** [reply]
Yes, there is no problem with the function of display the highlighted area. 

After rdkit.js generates the svg text, I want to further design some interactions in the webpage. Native atoms and bonds are too narrow to be selected. So I highlight all atoms and bonds. Binding the click event to the highlighted element (change the color when clicked), which is fast (this operation is no need to request wasm, just chang element color of SVG), so as to achieve the purpose of selecting atoms and bonds by clicking on the highlighted area. But the highlighted element doesn't label which bond or atom it comes from. I can only map by their coordinates at this stage. Poor interactive experience due to the influence of map operations and coordinate conversion of float numbers.

**So when rdkit returns svg highlight elements with its source (index of atoms or bonds), which will make web-coding more happier.**

Thank you ! @greglandrum

**Reply by greglandrum** [reply]
@ShangChien What I was trying to say is that when you highlight atoms/bonds you *do* end up with information about which atoms/bonds are involved in the SVG. For example, here's the SVG for a molecule with two atoms and the connecting bond highlighted:
```
<?xml version='1.0' encoding='iso-8859-1'?>
<svg version='1.1' baseProfile='full'
              xmlns='http://www.w3.org/2000/svg'
                      xmlns:rdkit='http://www.rdkit.org/xml'
                      xmlns:xlink='http://www.w3.org/1999/xlink'
                  xml:space='preserve'
width='450px' height='300px' viewBox='0 0 450 300'>
<!-- END OF HEADER -->
<rect style='opacity:1.0;fill:#FFFFFF;stroke:none' width='450.0' height='300.0' x='0.0' y='0.0'> </rect>
<path class='bond-1 atom-1 atom-2' d='M 203.8,97.1 L 387.2,202.9' style='fill:none;fill-rule:evenodd;stroke:#FF7F7F;stroke-width:45.2px;stroke-linecap:butt;stroke-linejoin:miter;stroke-opacity:1' />
<ellipse cx='203.8' cy='97.1' rx='42.3' ry='42.3' class='atom-1'  style='fill:#FF7F7F;fill-rule:evenodd;stroke:#FF7F7F;stroke-width:1.0px;stroke-linecap:butt;stroke-linejoin:miter;stroke-opacity:1' />
<ellipse cx='387.2' cy='202.9' rx='42.3' ry='42.3' class='atom-2'  style='fill:#FF7F7F;fill-rule:evenodd;stroke:#FF7F7F;stroke-width:1.0px;stroke-linecap:butt;stroke-linejoin:miter;stroke-opacity:1' />
<path class='bond-0 atom-0 atom-1' d='M 20.5,202.9 L 203.8,97.1' style='fill:none;fill-rule:evenodd;stroke:#000000;stroke-width:2.0px;stroke-linecap:butt;stroke-linejoin:miter;stroke-opacity:1' />
<path class='bond-1 atom-1 atom-2' d='M 203.8,97.1 L 387.2,202.9' style='fill:none;fill-rule:evenodd;stroke:#000000;stroke-width:2.0px;stroke-linecap:butt;stroke-linejoin:miter;stroke-opacity:1' />
</svg>
```
You can see that the first three elements (the highlights) all have class information which indicates which atoms/bonds they are associated with.

Isn't this what you're asking for?

**Reply by ShangChien** [reply]
Yeah that's what i want. 
I just updated to the latest 2022.03.1version from 2021.09 and it did solve my previous problem.
Good jobs ! Thank you !

**Reply by MichelML** [reply]
Learned something today! @ShangChien feel free to mark this as answered :+1:
